packages/language-server/src/hover.ts

```diff
@@ -0,0 +1,255 @@
+import {
+  TextDocumentPositionParams,
+  TextDocument,
+  Hover,
+} from "vscode-languageserver";
+import { parse, DocumentCstNode } from "@xml-tools/parser";
+import {
+  buildAst,
+  XMLAttribute,
+  XMLElement,
+  XMLAstVisitor,
+  accept,
+  XMLDocument,
+  XMLToken,
+} from "@xml-tools/ast";
+import { find } from "lodash";
+import {
+  flattenEvents,
+  flattenProperties,
+  flattenAssociations,
+  xmlToFQN,
+  ui5NodeToFQN,
+  flattenAggregations,
+} from "@ui5-language-assistant/logic-utils";
+import {
+  UI5Event,
+  UI5Prop,
+  UI5Association,
+  UI5Class,
+  UI5Aggregation,
+  UI5SemanticModel,
+  BaseUI5Node,
+  UI5Namespace,
+  UI5EnumValue,
+} from "@ui5-language-assistant/semantic-model-types";
+import { getNodeDocumentation } from "./documentation";
+
+interface XMLElementHover {
+  kind: "XMLElementHover";
+  astNode: XMLElement;
+}
+interface XMLKeyAttributeHover {
+  kind: "XMLKeyAttributeHover";
+  astNode: XMLAttribute;
+}
+interface XMLValueAttributeHover {
+  kind: "XMLValueAttributeHover";
+  astNode: XMLAttribute;
+}
+export type XMLHoverContext =
+  | XMLElementHover
+  | XMLKeyAttributeHover
+  | XMLValueAttributeHover;
+
+class HoverContextVisitor implements XMLAstVisitor {
+  public hoverContext: XMLHoverContext | undefined;
+  constructor(public chosenOffset: number | undefined) {}
```

Comment:
`chosenOffset` could be called just `offset` as we can understand the `hover` part from the context (inside an HoverVisitor).